### PR TITLE
[Spring] Private Storage Access support

### PR DESCRIPTION
**Pull request by haoozhang**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/spring/azext_spring/_params.py
Comment (allxiao) [inline src/spring/azext_spring/_params.py:98]: `underlying` --> `artifact` or `backend`?

File: src/spring/azext_spring/_params.py
Comment (allxiao) [inline src/spring/azext_spring/_params.py:269]: same question for the word `underlying`

File: src/spring/azext_spring/custom.py
Comment (allxiao) [inline src/spring/azext_spring/custom.py:119]: space between `=`. Same for other occurrences.

please further check the code styles of this PR.

File: src/spring/azext_spring/custom.py
Comment (allxiao) [inline src/spring/azext_spring/custom.py:127]: Is `None` the same as `null`, or more specifically, does `None` has the effect of "do not change" rather than "disable"?

File: src/spring/HISTORY.md
Comment (jiec-msft) [inline src/spring/HISTORY.md:3]: Use 1.21.0 for new feature. `1.20.1` was a mistake.

File: src/spring/azext_spring/_client_factory.py
Comment (jiec-msft) [inline src/spring/azext_spring/_client_factory.py:7]: Backend is not fully released, I think we cannot merge this PR

Comment (haoozhang) [reply src/spring/azext_spring/_client_factory.py:7]: thanks for your reminder, will merge it after code release.

File: src/spring/azext_spring/_params.py
Comment (haoozhang) [reply src/spring/azext_spring/_params.py]: updated.

File: src/spring/azext_spring/_params.py
Comment (haoozhang) [reply src/spring/azext_spring/_params.py]: updated.

File: src/spring/azext_spring/custom.py
Comment (haoozhang) [reply src/spring/azext_spring/custom.py]: updated.

**Comment by jiec-msft** [comment]
Since this PR changed the CLI version, it will release many "hiden" PRs merged after `1.20.1` and yours. Including SDK and default API change. Suggest to build a whl package and run the CLI E2E test according to [wiki](https://dev.azure.com/msazure/AzureDMSS/_wiki/wikis/AzureDMSS.wiki/379448/Integration-test-CLI)

Note, since current CLI code base for Spring extension requires CLI core 2.56.0+, so you can check this test [pipeline](https://msazure.visualstudio.com/AzureDMSS/_build/results?buildId=91145578&view=results), it upgrades the CLI core version.

**Comment by haoozhang** [comment]
> Since this PR changed the CLI version, it will release many "hiden" PRs merged after `1.20.1` and yours. Including SDK and default API change. Suggest to build a whl package and run the internal CLI E2E test according to [wiki](https://dev.azure.com/msazure/AzureDMSS/_wiki/wikis/AzureDMSS.wiki/379448/Integration-test-CLI)
> 
> Note, since current CLI code base for Spring extension requires CLI core 2.56.0+, so you can check this test [pipeline](https://msazure.visualstudio.com/AzureDMSS/_build/results?buildId=91145578&view=results), it upgrades the CLI core version.

CLI E2E test result: https://msazure.visualstudio.com/AzureDMSS/_build/results?buildId=92189362&view=results

**Review by allxiao — APPROVED** [review]

**Comment by haoozhang** [comment]
> Please fix conflicting files.

fixed, please help to review again, thanks!

File: src/spring/HISTORY.md
Comment (yanzhudd) [inline src/spring/HISTORY.md:13]:
```suggestion
* Update commands `az spring create` and `az spring update` to include a new argument `--enable-private-storage-access` to set whether private network access to backend storage in vnet injection instance.
```

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**
